pkg/cli/userfiletable_test.go

```diff
@@ -80,6 +81,156 @@ func Example_userfile_upload() {
 	// successfully uploaded to userfile://defaultdb.public.userfiles_root/test/file3.csv
 }
 
+func Example_userfile_upload_recursive() {
+	if err := runExampleUserfileUploadRecursive(); err != nil {
+		fmt.Println(err)
+	}
+}
+
+func runExampleUserfileUploadRecursive() (err error) {
+	c := NewCLITest(TestCLIParams{})
+	defer c.Cleanup()
+
+	tmpDir, err := ioutil.TempDir("", testUserfileUploadTempDirPrefix)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = errors.CombineErrors(err, os.RemoveAll(tmpDir))
+	}()
+	testDir := filepath.Join(tmpDir, "testdir/")
+	if err = os.MkdirAll(filepath.Join(testDir, "d1/d11/d111/d1111"), 0755); err != nil {
+		return err
+	}
+	if err = os.MkdirAll(filepath.Join(testDir, "d2/d21"), 0755); err != nil {
+		return err
+	}
+
+	filesRelativePaths := []string{
+		"d1/" + testTempFilePrefix + "f1.csv",
+		"d1/d11/" + testTempFilePrefix + "f2.csv",
+		"d1/d11/" + testTempFilePrefix + "f3.csv",
```

Comment:
don't know if writing to `d1/d11/` was intentional or not, but this got me thinking, the user would not see`d1/d11/d111` since it has no files. Since userfile is not a real fs we do not have a concept of empty directories. Maybe we should call this out in the help message, and say all non-empty directories will be uploaded?